index.js

```diff
@@ -1,89 +1,91 @@
-   * @returns {this}
-   */
-  append(statement) {
-    if (statement instanceof SQLStatement) {
-      this.strings[this.strings.length - 1] += statement.strings[0]
-      this.strings.push.apply(this.strings, statement.strings.slice(1))
-      const list = this.values || this.bind
-      list.push.apply(list, statement.values)
-    } else {
-      this.strings[this.strings.length - 1] += statement
-    }
-    return this
-  }
+    /** Returns the SQL Statement for mysql */
+    get sql() {
+      return strings.join('?')
+    },
 
-  /**
-   * Use a prepared statement with Sequelize.
-   * Makes `query` return a query with `$n` syntax instead of `?`  and switches the `values` key name to `bind`
-   * @param {boolean} [value=true] value If omitted, defaults to `true`
-   * @returns this
-   */
-  useBind(value) {
-    if (value === undefined) {
-      value = true
-    }
-    if (value && !this.bind) {
-      this.bind = this.values
-      delete this.values
-    } else if (!value && this.bind) {
-      this.values = this.bind
-      delete this.bind
-    }
-    return this
-  }
+    /** Returns the SQL Statement for node-postgres */
+    get text() {
+      return strings.reduce((p, c, i) => `${p}$${i}${c}`)
+    },
 
-  /**
-   * @param {string} name
-   * @returns {this}
-   */
-  setName(name) {
-    this.name = name
-    return this
-  }
-}
+    /** Returns values for others */
+    get values() {
+      return this.bound ? void 0 : values
+    },
 
-/** Returns the SQL Statement for mysql */
-Object.defineProperty(SQLStatement.prototype, 'sql', {
-  enumerable: true,
-  get() {
-    return this.strings.join('?')
-  },
-})
+    // Methods
 
-/**
- * @param {string[]} strings
- * @param {...any} values
- * @returns {SQLStatement}
- */
-function SQL(strings) {
-  return new SQLStatement(strings.slice(0), Array.from(arguments).slice(1))
+    /**
+     * @param {SQLStatement|string} statement
+     * @returns {this}
+     */
+    append(statement) {
+      if (typeof statement === 'string') {
+        strings[strings.length - 1] += statement
+      } else {
+        strings[strings.length - 1] += statement.strings[0]
+        strings.push(...statement.strings.slice(1))
+        values.push(...statement.values)
+      }
+      return this
+    },
+
+    /**
+     * @param {string} name
+     * @returns {this}
+     */
+    setName(name) {
+      this.name = name
+      return this
+    },
+
+    /**
+     * Use a prepared statement with Sequelize.
+     * Makes `query` return a query with `$n` syntax instead of `?` and
+     * uses `bind` instead of `values`
+     * @param {boolean} [bound=true] bound If omitted, defaults to `true`
+     * @returns this
+     */
+    useBind(bound = true) {
+      this.bound = bound
+      return this
+    },
+
+    // Properties
+
+    /** Default bound value as false */
+    bound: false,
+
+    /** Default name as empty string */
+    name: '',
+
+    /** Initial strings */
+    strings,
```

Comment:
These used to be private, is there a reason to expose them?

Reply:
strings property has never been private in your code, you access it on `append` already, and you set it as configurable, enumerable, and writable, within the `constructor`, same as here.

Reply:
I consider a proeprty "private" when it is not part of the public (documented) API. `append()` is part of the class so it can use private properties. Check the documentation which mentions it as private: http://node-sql-template-strings.surge.sh/classes/sqlstatement.html#strings
Before in a class we needed it as a property, but now `strings` is in scope, so this wouldn't be needed anymore.

Reply:
it's still needed via `append`, so either I use a `WeakMap` to have real privates, or it stays documented as private as it was before.

This literally doesn't change anything to the object, its property descriptor, or the accessibility of `strings`. I could state private as comment, but is exactly as "_private_" as before.

Reply:
As extra note: accessing `strings` is necessary for backward compatibility, so there's literally no point, since it's already documented as private, to not carry along the property s it was done before.

Reply:
Why would WeakMaps be needed? Isn't `strings` in scope for each new instance?

Reply:
`append(object)` ... how do you get `object.strings` there ?